api/agent/agent.go

```diff
@@ -231,6 +238,11 @@ func (a *agent) Submit(callI Call) error {
 	ctx = opentracing.ContextWithSpan(context.Background(), span)
 	call.End(ctx, err)
 
+	err = a.fireAfterCall(ctx, call.Model())
```

Comment:
would shove this into `End`, maybe we need to make it return an `error` again (fine w/ me)